src/helm/proxy/clients/critique_client.py

```diff
@@ -209,3 +217,239 @@ def make_critique_request(self, request: CritiqueRequest) -> CritiqueRequestResu
+                        f"{project.params['instructions']}"
+                    ) from err
+                elif project.type != TaskType.TextCollection.value:
+                    raise RuntimeError(
+                        f"Project {template.name} already exists with different task_type: '{project.type}' instead of '{TaskType.TextCollection.value}'"
+                    ) from err
+            return {"name": project.name}
+
+        with _scale_cache_lock:
+            # Example cache key:
+            # {
+            #   "name": "some_name",
+            #   "instructions": "some_instructions"
+            # }
+            #
+            # Example cache value:
+            # {"name": "some_name"}
+            project_response, is_cached = self._cache.get(
+                {"name": template.name, "instructions": template.instructions}, create_scale_project
+            )
+        project_name = project_response["name"]
+        if is_cached:
+            hlog(f"Reusing existing Scale project: {project_name}")
+        else:
+            hlog(f"Creating new Scale project: {project_name}")
+        return project_name
+
+    def _interpolate_fields(self, text: str, fields: Dict[str, str]) -> str:
+        for field_name, field_value in fields.items():
+            text = text.replace(f"{{{field_name}}}", field_value)
+        return text
+
+    def _critique_question_to_scale_field(self, question: CritiqueQuestionTemplate, fields: Dict[str, str]):
+        if question.question_type == "multiple_choice" or question.question_type == "checkbox":
+            return {
+                "type": "category",
+                "field_id": question.name,  # This must be unique, so we use the question name
+                "title": question.name,
+                "description": self._interpolate_fields(question.text, fields),
+                "choices": [{"label": option, "value": option} for option in question.options],
+            }
+        else:
+            raise ValueError(f"Unsupported question type {question.question_type}")
+
+    def _get_or_create_scale_task(
+        self, project_name: str, template: CritiqueTaskTemplate, fields: Dict[str, str]
+    ) -> str:
+        """Get or create a task on Scale and return the Scale task ID."""
+
+        def create_scale_task() -> Dict[str, str]:
+            """
+            Creates a Scale Task (which is one single question from a CritiqueQuestionTemplate)
+            Returns the Scale Task ID.
+            """
+
+            # We create a unique_id for the task so that we can reuse it if it already exists
+            # It contains the same information as the task itself (like the cache key)
+            # This is redundant with the cache but it's a good safety net
+            # NOTE: Technically, sha512 could have collisions but it's unlikely.
+            unique_id: str = sha512(
+                str({"project": project_name, "task": unstructure(template), "fields": fields}).encode()
```

Comment:
Follow what we do for [`cache.request_to_key()`](https://github.com/stanford-crfm/helm/blob/14407c5147f5307401be27339d7b5be153d13f71/src/helm/common/cache.py#L22), so that we don't depend on iteration order of fields.

I'd also suggest making `{"project": project_name, "task": unstructure(template), "fields": fields}` a named variable, since it gets used for the cache key also (you can define it outside the method and capture in closure scope).

I would also be okay with leaving `unique_id` unset.